frontend/public/components/nav/section.tsx

```diff
@@ -91,16 +94,33 @@ export const NavSection = connect(navSectionStateToProps)(
       this.setState({isOpen: expandState});
     }
 
+    getNavItems(perspective: string, section: string) {
+      const defaultPerspective = _.find(
+        plugins.registry.getPerspectives(),
+        p => p.properties.default
+      );
+      const isDefaultPerspective =
+        defaultPerspective && perspective === defaultPerspective.properties.id;
+      return plugins.registry.getNavItems().filter(
+        item =>
+          // check if the item is contributed to the current perspective,
+          // or if no perspective specified, are we in the default perspective
+          (item.properties.perspective === perspective ||
+            (!item.properties.perspective && isDefaultPerspective)) &&
+          item.properties.section === section
+      );
+    }
+
     getPluginNavItems = memoize(
-      (section) => plugins.registry.getNavItems(section)
+      (perspective, section) => this.getNavItems(perspective, section)
```

Comment:
Nit-pick: this could be shortened to

```tsx
getPluginNavItems = memoize(this.getNavItems);
```

Also, we're currently lacking types for `memoize-one` library because IIRC the ones at [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/memoize-one) were messed up.

As a result, the return type of `getPluginNavItems` becomes `any` so we should fix that in future, maybe force-type the result to `NavItem[]`.

Reply:
Fixed.